Solutions/Marain.Workflow.Abstractions/Marain/Workflows/WorkflowEngine.cs

```diff
@@ -70,6 +70,10 @@ public Task<WorkflowInstance> StartWorkflowInstanceAsync(StartWorkflowInstanceRe
             string instancePartitionKey = null,
             IDictionary<string, string> context = null)
         {
+            // We need to clone the original context here in case it's modified by any of the code which creates the
```

Comment:
This is one reason I'm favouring Immutable data structures for everything. You never have this problem.

Reply:
Agreed.